core/src/main/java/org/pentaho/di/core/logging/KettleLogStore.java

```diff
@@ -0,0 +1,278 @@
+ *
+ * Copyright (C) 2002-2020 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.core.logging;
+
+import java.io.PrintStream;
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.pentaho.di.core.Const;
+import org.pentaho.di.core.util.EnvUtil;
+
+public class KettleLogStore {
+
+  public static PrintStream OriginalSystemOut = System.out;
+  public static PrintStream OriginalSystemErr = System.err;
+
+  private static KettleLogStore store;
+
+  private LoggingBuffer appender;
+
+  private Timer logCleanerTimer;
+
+  private static AtomicBoolean initialized = new AtomicBoolean( false );
+
+  private static LogChannelInterfaceFactory logChannelInterfaceFactory = new LogChannelFactory();
+
+  public static LogChannelInterfaceFactory getLogChannelInterfaceFactory() {
+    return logChannelInterfaceFactory;
+  }
+
+  public static void setLogChannelInterfaceFactory( LogChannelInterfaceFactory logChannelInterfaceFactory ) {
+    KettleLogStore.logChannelInterfaceFactory = logChannelInterfaceFactory;
+  }
+
+  /**
+   * Create the central log store with optional limitation to the size
+   *
+   * @param maxSize
+   *          the maximum size
+   * @param maxLogTimeoutMinutes
+   *          The maximum time that a log line times out in Minutes.
+   */
+  private KettleLogStore( int maxSize, int maxLogTimeoutMinutes, boolean redirectStdOut, boolean redirectStdErr ) {
+    this.appender = new LoggingBuffer( maxSize );
+    replaceLogCleaner( maxLogTimeoutMinutes );
+
+    if ( redirectStdOut ) {
+      System.setOut( new LoggingPrintStream( OriginalSystemOut ) );
+    }
+
+    if ( redirectStdErr ) {
+      System.setErr( new LoggingPrintStream( OriginalSystemErr ) );
+    }
+  }
+
+  public void replaceLogCleaner( final int maxLogTimeoutMinutes ) {
+    if ( logCleanerTimer != null ) {
+      logCleanerTimer.cancel();
+    }
+    logCleanerTimer = new Timer( true );
+
+    TimerTask timerTask = new TimerTask() {
+      @Override
+      public void run() {
+
+        if ( maxLogTimeoutMinutes > 0 ) {
+          long minTimeBoundary = new Date().getTime() - maxLogTimeoutMinutes * 60 * 1000;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).